lib/util/exec/index.ts

```diff
@@ -84,28 +84,34 @@ export async function exec(
     cwd,
   };
 
-  // TODO: simplify after callers' refactoring
-  const singleCommand = typeof cmd === 'string' ? cmd : cmd.join(' && ');
-  let pExecCommand = singleCommand.startsWith('docker run')
-    ? singleCommand
-    : `bash -l -c "${singleCommand.replace(/"/g, '\\"')}"`;
-
+  let commands = typeof cmd === 'string' ? [cmd] : cmd;
   if (docker) {
     const dockerOptions = {
       ...docker,
       cwd,
       envVars: dockerEnvVars(extraEnv, childEnv),
     };
 
-    pExecCommand = dockerCmd(pExecCommand, dockerOptions);
+    let singleCommand = commands.join(' && ');
+    singleCommand = `bash -l -c "${singleCommand.replace(/"/g, '\\"')}"`;
+    singleCommand = dockerCmd(singleCommand, dockerOptions);
+    commands = [singleCommand];
+  }
+
+  let res: ExecResult | null = null;
+  for (const pExecCommand of commands) {
+    res = await pExec(pExecCommand, pExecOptions);
   }
 
-  const res = await pExec(pExecCommand, pExecOptions);
   const duration = hrtime(startTime);
   const seconds = Math.round(duration[0] + duration[1] / 1e9);
-  logger.debug(
-    { cmd, seconds, stdout: res.stdout, stderr: res.stderr },
-    'exec completed'
-  );
+
+  if (res) {
```

Comment:
I think this timing should be moved inside the for loop so that we capture every command and not not just the last one